CRM/Financial/Form/Payment.php

```diff
@@ -109,17 +109,17 @@ public function setDefaultValues() {
    *
    * @param int $paymentProcessorID
    * @param string $region
+   * @param CRM_Core_Form $form
    */
-  public static function addCreditCardJs($paymentProcessorID = NULL, $region = 'billing-block') {
+  public static function addCreditCardJs($paymentProcessorID = NULL, $region = 'billing-block', $form) {
     $creditCards = CRM_Financial_BAO_PaymentProcessor::getCreditCards($paymentProcessorID);
     $creditCardTypes = CRM_Core_Payment_Form::getCreditCardCSSNames($creditCards);
+    // workaround for CRM-1364 and CRM-20264 and CRM-20516
+    $form->assign('creditCardTypes', json_encode($creditCardTypes));
```

Comment:
@JoeMurray @eileenmcnaughton i have decided to assign this to a Smarty variable here the reasoning is for some reason the addScript can only work for wordpress i think in html-header but that doesn't work for the popup forms. So consequently i then set the setting `CRM.config.creditCardTypes` through new Line 30 in the BillingBlock.tpl. It would appear from the reports that BillingBlock.js does at least get loaded in Wordpress